testsuite/tests/shadow_include/artificial.ml

```diff
@@ -0,0 +1,51 @@
+(* TEST
+   * expect
+*)
+
+module Foo : sig
+  type t
+
+  module Bar : sig
+    type t
+  end
+
+  val to_ : t -> Bar.t
+  val from: Bar.t -> t
+end = struct
+  type t
+
+  module Bar = struct
+    type nonrec t = t
+  end
+
+  let to_ x = x
+  let from x = x
+end
+;;
+[%%expect{|
+module Foo :
+  sig
+    type t
+    module Bar : sig type t end
+    val to_ : t -> Bar.t
+    val from : Bar.t -> t
+  end
+|}]
+
+module Extended = struct
+  include Foo
+  module Bar = struct
+    include Bar
+    let int = 42
+  end
+end
+;;
+[%%expect{|
+module Extended :
+  sig
+    type t = Foo.t
+    val to_ : t -> Foo.Bar.t
+    val from : Foo.Bar.t -> t
+    module Bar : sig type t = Foo.Bar.t val int : int end
+  end
+|}]
```

Comment:
Here I would add a test for the associated module type that a user would put in the signature:
```
module type Extended_sig = sig
  include module type of struct include Foo end
  module Bar : sig
    include module type of Bar
    val int : int
  end
end
```

Reply:
Indeed, done.